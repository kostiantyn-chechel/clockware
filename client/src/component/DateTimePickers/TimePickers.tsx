import * as React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { nowTimePlus } from "../../helpers/dateTime";

const useStyles = makeStyles((theme) => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
}));

interface ItimePickers {
    onChange(event: React.ChangeEvent<{ value: string; }>): void;
}

const TimePickers: React.FC<ItimePickers> = (props) => {
    const classes = useStyles();

    return (
        <div className={classes.container}>
            <TextField
                onChange={props.onChange}
                id="time"
                label="Время:"
                type="time"
                defaultValue={nowTimePlus()}
                className={classes.textField}
                InputLabelProps={{
                    shrink: true,
                }}
                inputProps={{
                    step: 3600, // 60 min
                }}
            />
        </div>
    );
};

export default TimePickers;