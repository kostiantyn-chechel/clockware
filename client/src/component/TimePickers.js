import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';

const useStyles = makeStyles((theme) => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
}));

export default function TimePickers(props) {
    const classes = useStyles();

    return (
        // <form className={classes.container} noValidate>
        <div className={classes.container}>
            <TextField
                onChange={props.onChange}
                id="time"
                label="Время:"
                type="time"
                defaultValue="10:00"
                className={classes.textField}
                InputLabelProps={{
                    shrink: true,
                }}
                inputProps={{
                    step: 1800, // 30 min
                }}
            />
        </div>
    );
}