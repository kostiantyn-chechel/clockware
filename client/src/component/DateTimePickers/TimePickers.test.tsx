import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import TimePickers from './TimePickers';

jest.mock('../../helpers/dateTime', () => ({
    nowTimePlus: () => '13:00',
}));

describe('Test <TimePickers>', () => {
    const onChangeMock = jest.fn();

    const Component = <TimePickers onChange={onChangeMock}/>;
    const ShallowComponent = shallow(Component);

    test('TimePickers component should exists', () => {
        expect(ShallowComponent).toBeDefined();
    });

    test("ToMatchSnapshot TimePickers with default date", () => {
        expect(renderer.create(Component).toJSON()).toMatchSnapshot();
    });

    test("DatePickers changes value after click", () => {
        const MountComponent = mount(Component);
        const newTime = '15:00';
        MountComponent.find("input").simulate("change", {
            target: { value: newTime },
        });

        expect(onChangeMock).toHaveBeenCalledTimes(1);

        // expect(onChangeMock).toHaveBeenCalledWith(newTime); // ???

    });

});