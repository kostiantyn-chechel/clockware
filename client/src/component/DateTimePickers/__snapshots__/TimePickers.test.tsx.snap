// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TimePickers> ToMatchSnapshot TimePickers with default date 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="time"
      id="time-label"
    >
      Время:
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        defaultValue="13:00"
        disabled={false}
        id="time"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        step={3600}
        type="time"
      />
    </div>
  </div>
</div>
`;
